src/plugins/numberline/numberline-tile.tsx

```diff
@@ -35,61 +42,222 @@ export const NumberlineToolComponent: React.FC<ITileProps> = observer((props) =>
 
-  //----------------- Create Numberline Axis  --------------
-  useEffect(()=>{
-    // Construct axis
-    const linearScale = scaleLinear()
-    .domain([numberlineDomainMin, numberlineDomainMax])
-    .range([0, axisWidth]);
-    const axis = axisBottom(linearScale).tickSizeOuter(0);
-    const numOfTicks = numberlineDomainMax - numberlineDomainMin;
-    axis.ticks(numOfTicks);
-    (select(`.${axisClass}`) as d3.Selection<SVGSVGElement, unknown, HTMLElement, unknown>).call(axis);
-
-    // After the axis is drawn, customize "x = 0 tick"
-    selectAll("g.num-axis g.tick line")
+  //--------------------  SVG Ref to Numberline & SVG / Mouse State ---------------------------------
+  const svgRef = useRef<SVGSVGElement | any>(null);
+  const svg = select(svgRef.current);
+  const svgNode = svg.node();
+  const axisRef = useRef<SVGGElement | any>(null);
+  const axis = select(axisRef.current);
+  const [mouseIsDragging, setMouseIsDragging] = useState(false); //only within numberline
+  const [mousePosX, setMousePosX] = useState(0); //used to retrigger useEffect
+
+  /* ============================= [ Handlers/Utility Functions ]  =============================== */
+
+  const mouseInBoundingBox = (e: Event) => {
+    const pos = pointer(e, svgNode);
+    const xPos = pos[0];
+    const yPos = pos[1];
+    const yTopBound = yMidPoint + 15;
+    const yBottomBound = yMidPoint - 15;
+    const isBetweenYBounds = (yPos >= yBottomBound && yPos <= yTopBound);
+    const isBetweenXBounds = (xPos >= 0 && xPos <= axisWidth);
+    if (isBetweenYBounds && isBetweenXBounds ){
+      content.isMouseHoverOverPoint(xPos, axisWidth); //detect if hovered over an existing point
+      setMousePosX(xPos);
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const handleClickCreatePoint = (e: Event) => {
+    const pos = pointer(e, svgNode);
+    const xPos = pos[0];
+    const xValue = xScale.invert(xPos);
+    const newPoint = {xValue};
+    content.createNewPoint(newPoint);
+  };
+
+  const handleMouseClick = (e: Event) => {
+    if (!content.isHoveringOverPoint){
+      if (mouseInBoundingBox(e)){
+
+        handleClickCreatePoint(e); //only create point if we are not hovering over a point and within bounding box
+      }
+    } else{
+      content.toggleIsSelected(content.indexOfPointHovered);
+    }
+  };
+
+  const givenIDFindIndexPoint = (id: string)=> {
+    return content.axisPoints.findIndex((p)=> p.id === id);
+  };
+
+  const handleDrag = drag<SVGCircleElement, PointObjectModelType>()
+  .on('drag', (e, p) => {
+      if (mouseInBoundingBox(e)){
+        setMouseIsDragging(true);
+        const idxPointDragged = givenIDFindIndexPoint(p.id);
+        content.toggleIsSelected(idxPointDragged);//make the dragged point as the only one selected.
+        const oldAxisWidth = axisWidth;
+        const newAxisWidth = svgNode.getBoundingClientRect().width;
+        const isAxisResized = !!(Math.abs(oldAxisWidth - newAxisWidth) > 2);
```

Comment:
Is the `2` here just to make sure the axis changed a significant amount?

Reply:
And the `!!` is unnecessary since the result of `>` is already boolean.